src/server/pps/cmds/cmds.go

```diff
@@ -241,29 +244,27 @@ $ {{alias}} foo@XXX -p bar -p baz`,
 				return err
 			}
 			defer c.Close()
-
-			jobInfos, err := c.FlushJobAll(commits, pipelines)
-			if err != nil {
-				return err
+			var writer *tabwriter.Writer
+			if !raw {
+				writer = tabwriter.NewWriter(os.Stdout, pretty.JobHeader)
 			}
-
-			if raw {
-				e := encoder(output)
-				for _, jobInfo := range jobInfos {
-					if err := e.EncodeProto(jobInfo); err != nil {
+			if err := c.FlushJob(commits, pipelines, func(ji *ppsclient.JobInfo) error {
+				if raw {
+					e := encoder(output)
```

Comment:
Is there any overhead to repeatedly cresting an encoder?

Reply:
Very minimal, it's basically just a dumb struct, which when gets some options applied to it. But I'll move it outside anyways since who knows if that'll change, and might as well do the faster thing if there's no added complexity.